Datastore controller will retry single file queries

acron0 opened the pull request:
This is to avoid the 'too quick' bug following the upload of a file -
may have to up the # of tries or add a small sleep if it doesn't fix
it. Will now also display an error message once the tries are exhausted.

elisehuard commented on src/cljs/witan/ui/services/data.cljs:
```diff
@@ -207,8 +184,12 @@
-  (doseq [f forecast-versions]
-    (put-item-into-db! f :forecast))
+  (let [latest-forecast (first forecast-versions)
+        older-forecasts (map #(assoc % :descendant-id (:version-id latest-forecast)) (rest forecast-versions))]
```
nice one, makes sense

acron0 replied:
This means that 'descendant-id' is really "id of the latest forecast" but that change is already reflected when 'expanding' so that's fine.

elisehuard commented:
:shipit:

Commit:
Datastore controller will retry single file queries

This is to avoid the 'too quick' bug following the upload of a file -
may have to up the # of tries or add a small sleep if it doesn't fix
it. Will now also display an error message once the tries are exhausted.

Commit:
Remove `do`

otfrom commented:
:shipit:
